presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java

```diff
@@ -556,24 +887,78 @@ public static String getTemporaryStagingDirectoryPath(ConnectorSession session)
         return session.getProperty(TEMPORARY_STAGING_DIRECTORY_PATH, String.class);
     }
 
-    public static boolean isPreloadSplitsForGroupedExecution(ConnectorSession session)
+    public static String getTemporaryTableSchema(ConnectorSession session)
     {
-        return session.getProperty(PRELOAD_SPLITS_FOR_GROUPED_EXECUTION, Boolean.class);
+        return session.getProperty(TEMPORARY_TABLE_SCHEMA, String.class);
     }
 
-    public static boolean isWritingStagingFilesEnabled(ConnectorSession session)
+    public static HiveStorageFormat getTemporaryTableStorageFormat(ConnectorSession session)
     {
-        return session.getProperty(WRITING_STAGING_FILES_ENABLED, Boolean.class);
+        return session.getProperty(TEMPORARY_TABLE_STORAGE_FORMAT, HiveStorageFormat.class);
     }
 
-    public static String getTemporaryTableSchema(ConnectorSession session)
+    public static HiveCompressionCodec getTemporaryTableCompressionCodec(ConnectorSession session)
     {
-        return session.getProperty(TEMPORARY_TABLE_SCHEMA, String.class);
+        return session.getProperty(TEMPORARY_TABLE_COMPRESSION_CODEC, HiveCompressionCodec.class);
     }
 
-    public static HiveStorageFormat getTemporaryTableStorageFormat(ConnectorSession session)
+    public static boolean shouldCreateEmptyBucketFilesForTemporaryTable(ConnectorSession session)
     {
-        return session.getProperty(TEMPORARY_TABLE_STORAGE_FORMAT, HiveStorageFormat.class);
+        return session.getProperty(TEMPORARY_TABLE_CREATE_EMPTY_BUCKET_FILES, Boolean.class);
+    }
+
+    public static boolean isUsePageFileForHiveUnsupportedType(ConnectorSession session)
+    {
+        return session.getProperty(USE_PAGEFILE_FOR_HIVE_UNSUPPORTED_TYPE, Boolean.class);
+    }
+
+    public static boolean isPushdownFilterEnabled(ConnectorSession session)
+    {
+        return session.getProperty(PUSHDOWN_FILTER_ENABLED, Boolean.class);
+    }
+
+    public static boolean isRangeFiltersOnSubscriptsEnabled(ConnectorSession session)
+    {
+        return session.getProperty(RANGE_FILTERS_ON_SUBSCRIPTS_ENABLED, Boolean.class);
+    }
+
+    public static boolean isAdaptiveFilterReorderingEnabled(ConnectorSession session)
+    {
+        return session.getProperty(ADAPTIVE_FILTER_REORDERING_ENABLED, Boolean.class);
+    }
+
+    public static int getVirtualBucketCount(ConnectorSession session)
+    {
+        int virtualBucketCount = session.getProperty(VIRTUAL_BUCKET_COUNT, Integer.class);
+        if (virtualBucketCount < 0) {
+            throw new PrestoException(INVALID_SESSION_PROPERTY, format("%s must not be negative: %s", VIRTUAL_BUCKET_COUNT, virtualBucketCount));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: virtualBucketCount (declared at line 932). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 934. Consider formatting this data with &#39;%d&#39; instead.